src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -67,6 +67,30 @@ internal class AD7DebugSession : DebugAdapterBase, IDebugPortNotify2, IDebugEven
 
         private static Guid s_guidFilterAllLocalsPlusArgs = new Guid("939729a8-4cb0-4647-9831-7ff465240d5f");
 
+        private static readonly HandleCollection<IDebugModule2> s_debugModules = new HandleCollection<IDebugModule2>();
```

Comment:
>HandleCollection [](start = 32, length = 16)

I apologize, but actually HandleCollection doesn't do a great job for your scenario since you need to convert IDebugModule2->`id`, but, currently at least, we don't have any operations that go the other way.

Probably a better fit it:
```
private int m_nextModuleHandle = 1;
private readonly Dictionary<IDebugModule2, int> m_moduleMap = new Dictionary<IDebugModule2, int>();

object RegisterDebugModule(IDebugModule2 debugModule)
{
    Debug.Assert(!m_moduleMap.Contains(debugModule));
    lock (m_moduleMap)
    {
        int moduleHandle = m_nextModuleHandle;
        m_moduleMap[debugModule] = moduleHandle;
        m_nextModuleHandle++;
        return moduleHandle;
    }
}
```

Reply:
The return type for RegisterDebugModule() is an object and not an int because of the Debug.Assert(), correct?

Reply:
The 'id' field of a Protocol module is an object, so I made it an object so that we could change `m_nextModuleHandle` to something else if we wanted to without changing anything else. But you could certainly leave it as an int if you like.